sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -238,33 +238,37 @@ class kernel_bundle_impl {
                        });
   }
 
-  bool has_specialization_constant(unsigned int SpecID) const noexcept {
+  bool has_specialization_constant(const char *SpecName) const noexcept {
     return std::any_of(MDeviceImages.begin(), MDeviceImages.end(),
-                       [SpecID](const device_image_plain &DeviceImage) {
+                       [SpecName](const device_image_plain &DeviceImage) {
                          return getSyclObjImpl(DeviceImage)
-                             ->has_specialization_constant(SpecID);
+                             ->has_specialization_constant(SpecName);
                        });
   }
 
-  void set_specialization_constant_raw_value(unsigned int SpecID,
-                                             const void *Value,
-                                             size_t ValueSize) {
+  void set_specialization_constant_raw_value(const char *SpecName,
+                                             const void *Value) noexcept {
     for (const device_image_plain &DeviceImage : MDeviceImages)
       getSyclObjImpl(DeviceImage)
-          ->set_specialization_constant_raw_value(SpecID, Value, ValueSize);
+          ->set_specialization_constant_raw_value(SpecName, Value);
   }
```

Comment:
Could you please add TODO saying to add support for spec constants which are not known to any device image?